src/main/scala/edu/berkeley/ce/rockslicing/BlockVTK.scala

```diff
@@ -0,0 +1,213 @@
+package edu.berkeley.ce.rockslicing
+
+import breeze.linalg
+import breeze.linalg.{DenseVector, DenseMatrix}
+
+object BlockVTK {
+
+  /**
+    * Calculates the rotation matrix to rotate the input plane (specified by its normal)
+    * to the desired orientation (specified by the desired normal)
+    * @param n_current: Current normal of plane
+    * @param n_desired: Desired new normal
+    * @return 3*3 rotation matrix
+    */
+  private def rotationMatrix(n_current: (Double, Double, Double), n_desired: (Double, Double, Double)):
+                             DenseMatrix[Double] = {
+    val n_c = DenseVector[Double](n_current._1, n_current._2, n_current._3)
+    val n_d = DenseVector[Double](n_desired._1, n_desired._2, n_desired._3)
+    if (math.abs(linalg.norm(linalg.cross(n_c,n_d))) > NumericUtils.EPSILON) {
+      val v = linalg.cross(n_c, n_d)
+      val s = linalg.norm(v)
+      val c = n_c dot n_d
+
+      val v_skew = DenseMatrix.zeros[Double](3,3)
+      v_skew(0,1) = -v(2)
+      v_skew(0,2) = v(1)
+      v_skew(1,0) = v(2)
+      v_skew(1,2) = -v(0)
+      v_skew(2,0) = -v(1)
+      v_skew(2,1) = v(0)
+
+      DenseMatrix.eye[Double](3) + v_skew + (v_skew * v_skew) * (1-c)/(s*s)
+    } else {
+      DenseMatrix.eye[Double](3)
+    }
+  }
+
+  /**
+    * Finds the center of a list of vertices - defined as the average coordinate of all the vertices in the list
+    * @param vertices List of vertices
+    * @return Coordinate of center of vertices
+    */
+  private def findCenter(vertices: Seq[(Double, Double, Double)]): (Double, Double, Double) = {
+    (vertices.map(_._1).sum / vertices.length.toDouble,
+      vertices.map(_._2).sum / vertices.length.toDouble,
+      vertices.map(_._3).sum / vertices.length.toDouble)
+  }
+
+  /**
+    * Compares pointA to pointB. If pointA is first relative to pointB rotating counter-clockwise about center,
+    * the method returns true. Input vectors should have same z-coordinate - comparison is based in 2-D
+    * @param pointA Point of interest
+    * @param pointB Point of comparison
+    * @param center Reference point for comparison
+    * @return Returns TRUE if pointA is first relative to pointB, FALSE otherwise
+    */
+  private def ccwCompare(pointA: (Double, Double, Double), pointB: (Double, Double, Double),
+                         center: (Double, Double, Double)): Boolean = {
+    // Check that points are in the same x-y plane
+    assert(math.abs(pointA._3 - pointB._3) < NumericUtils.EPSILON)
```

Comment:
Usually `assert` is used to check that a situation that should be impossible doesn't occur. It's not really supposed to be used to check if you got sensible input.

So here, you could throw an exception, which I don't think is very Scala-esque, or you could just return false and describe this case in the method's documentation. Or there could always be another approach I'm not thinking of. :)